pkg/seedadmissioncontroller/webhooks/admission/extensionresources/admission.go

```diff
@@ -16,266 +16,281 @@ package extensionresources
-			validateCreateResource: func(n, _ client.Object) field.ErrorList {
-				return druidvalidation.ValidateEtcd(n.(*druidv1alpha1.Etcd))
-			},
-			validateUpdateResource: func(n, o client.Object) field.ErrorList {
-				return druidvalidation.ValidateEtcdUpdate(n.(*druidv1alpha1.Etcd), o.(*druidv1alpha1.Etcd))
-			},
-		},
-
-		gvr("extensions"): {
-			newObject: func() client.Object { return new(extensionsv1alpha1.Extension) },
-			validateCreateResource: func(n, _ client.Object) field.ErrorList {
-				return validation.ValidateExtension(n.(*extensionsv1alpha1.Extension))
-			},
-			validateUpdateResource: func(n, o client.Object) field.ErrorList {
-				return validation.ValidateExtensionUpdate(n.(*extensionsv1alpha1.Extension), o.(*extensionsv1alpha1.Extension))
-			},
-		},
-
-		gvr("infrastructures"): {
-			newObject: func() client.Object { return new(extensionsv1alpha1.Infrastructure) },
-			validateCreateResource: func(n, _ client.Object) field.ErrorList {
-				return validation.ValidateInfrastructure(n.(*extensionsv1alpha1.Infrastructure))
-			},
-			validateUpdateResource: func(n, o client.Object) field.ErrorList {
-				return validation.ValidateInfrastructureUpdate(n.(*extensionsv1alpha1.Infrastructure), o.(*extensionsv1alpha1.Infrastructure))
-			},
-		},
-
-		gvr("networks"): {
-			newObject: func() client.Object { return new(extensionsv1alpha1.Network) },
-			validateCreateResource: func(n, _ client.Object) field.ErrorList {
-				return validation.ValidateNetwork(n.(*extensionsv1alpha1.Network))
-			},
-			validateUpdateResource: func(n, o client.Object) field.ErrorList {
-				return validation.ValidateNetworkUpdate(n.(*extensionsv1alpha1.Network), o.(*extensionsv1alpha1.Network))
-			},
-		},
-
-		gvr("operatingsystemconfigs"): {
-			newObject: func() client.Object { return new(extensionsv1alpha1.OperatingSystemConfig) },
-			validateCreateResource: func(n, _ client.Object) field.ErrorList {
-				return validation.ValidateOperatingSystemConfig(n.(*extensionsv1alpha1.OperatingSystemConfig))
-			},
-			validateUpdateResource: func(n, o client.Object) field.ErrorList {
-				return validation.ValidateOperatingSystemConfigUpdate(n.(*extensionsv1alpha1.OperatingSystemConfig), o.(*extensionsv1alpha1.OperatingSystemConfig))
-			},
-		},
-
-		gvr("workers"): {
-			newObject: func() client.Object { return new(extensionsv1alpha1.Worker) },
-			validateCreateResource: func(n, _ client.Object) field.ErrorList {
-				return validation.ValidateWorker(n.(*extensionsv1alpha1.Worker))
-			},
-			validateUpdateResource: func(n, o client.Object) field.ErrorList {
-				return validation.ValidateWorkerUpdate(n.(*extensionsv1alpha1.Worker), o.(*extensionsv1alpha1.Worker))
-			},
-		},
+// AddWebhooks add extension's valdation webhook to manager
+func AddWebhooks(mgr manager.Manager) error {
+	if err := builder.WebhookManagedBy(mgr).WithValidator(&backupBucketValidator{}).For(&extensionsv1alpha1.BackupBucket{}).Complete(); err != nil {
```

Comment:
Another suggestion: What about something like this:

```go
for obj, validator := range map[client.Object]admission.CustomValidator{
    &extensionsv1alpha1.BackupBucket{}: &backupBucketValidator{},
    // ...
} {
    if err := builder.WebhookManagedBy(mgr).WithValidator(validator).For(obj).Complete(); err != nil {
        return err
    }
}
```

Reply:
Just one look gives the idea that many webhooks are being added not very strong very opinion.

Reply:
Addressed in c0823b52f0a9ed4a2d6153de036df5de60edc962